client/components/language-picker/modal.jsx

```diff
@@ -0,0 +1,256 @@
+			}
+		}
+
+		return suggestedLanguages;
+	}
+
+	handleSearch = ( search ) => {
+		this.setState( { search } );
+	}
+
+	handleClick = ( selectedLanguageSlug ) => {
+		this.setState( { selectedLanguageSlug } );
+	}
+
+	handleSelectLanguage = () => {
+		const langSlug = this.state.selectedLanguageSlug;
+		this.props.onSelected( langSlug );
+		this.handleClose();
+	}
+
+	handleClose = () => {
+		this.props.onClose();
+	}
+
+	renderTabItems() {
+		const tabs = [ 'popular', '' ];
+
+		return map( tabs, filter => {
+			const selected = this.state.filter === filter;
+			const onClick = () => this.setState( { filter } );
+
+			return (
+				<SectionNavTabItem
+					key={ filter }
+					selected={ selected }
+					onClick={ onClick }
+				>
+					{ this.getFilterLabel( filter ) }
+				</SectionNavTabItem>
+			);
+		} );
+	}
+
+	renderLanguageList() {
+		const languages = this.getDisplayedLanguages();
+		const listClasses = classNames( 'language-picker__modal-list', {
+			// The language list has a different height when the 'suggested' list
+			// is also present. TODO: rewrite to flex!
+			'has-suggested-languages': this.state.suggestedLanguages
+		} );
+
+		return (
+			<div className={ listClasses }>
+				{ map( languages, this.renderLanguageItem ) }
+			</div>
+		);
+	}
+
+	renderLanguageItem = ( language ) => {
+		const isSelected = language.langSlug === this.state.selectedLanguageSlug;
+		const classes = classNames( 'language-picker__modal-text', {
+			'is-selected': isSelected
+		} );
+
+		return (
+			<div
+				className="language-picker__modal-item"
+				key={ language.langSlug }
+				onClick={ partial( this.handleClick, language.langSlug ) }
+			>
+				<span className={ classes }>{ language.name }</span>
+			</div>
+		);
+	}
+
+	renderSuggestedLanguages() {
+		const { suggestedLanguages } = this.state;
+		if ( ! suggestedLanguages ) {
+			return null;
+		}
+
+		return (
+			<div className="language-picker__modal-suggested">
+				<div className="language-picker__modal-suggested-label">
+					{ this.props.translate( 'Suggested languages: ' ) }
+				</div>
+				{ suggestedLanguages.map( language => {
+					const isSelected = language.langSlug === this.state.selectedLanguageSlug;
+					const classes = classNames( 'language-picker__modal-text', {
+						'is-selected': isSelected
+					} );
+
+					return (
+						<div
+							key={ language.langSlug }
+							className={ classes }
+							onClick={ partial( this.handleClick, language.langSlug ) }
+						>{ language.name }</div>
+					);
+				} ) }
+			</div>
+		);
+	}
+
+	render() {
+		const { isVisible, translate } = this.props;
+
+		if ( ! isVisible ) {
+			// Render nothing at all if the modal is not visible
+			// <Dialog isVisible={ false }> still renders a lot of useless elements
+			return null;
+		}
+
+		const buttons = [
+			{
+				action: 'cancel',
+				label: translate( 'Cancel' )
+			},
+			{
+				action: 'confirm',
+				label: translate( 'Select Language' ),
```

Comment:
Hi! I've found a possible matching string that has already been translated **60** times:
`translate( 'Select a language below' )` *ES Score: 10*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/language-chooser)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).